pkg/kubernetes/client/kubeclient_interface.go

```diff
@@ -137,6 +138,8 @@ type KubeClientConnector interface {
 	ListDatabaseEngines(ctx context.Context, namespace string) (*everestv1alpha1.DatabaseEngineList, error)
 	// GetDatabaseEngine returns database clusters by provided name.
 	GetDatabaseEngine(ctx context.Context, namespace, name string) (*everestv1alpha1.DatabaseEngine, error)
+	// Deployment returns a deployment.
+	Deployment(ctx context.Context, name, namespace string) (*appsv1.Deployment, error)
```

Comment:
I'm trying to follow [Effective Go](https://go.dev/doc/effective_go#Getters) where it recommends against using `Get` verb in a function name.
Similarly we have 
```
kubernetes.CoreV1().Pods(namespace).List(ctx, options)`
// instead of
kubernetes.GetCoreV1().GetPods(namespace).RunList(ctx, options)
```

I'm happy to rename it to `GetDeployment` if you feel this is not relevant to us.